lms/djangoapps/courseware/tests/test_masquerade.py

```diff
@@ -209,6 +233,26 @@ def create_user(self):
         return StaffFactory(course_key=self.course.id)
 
 
+class TestMasqueradeOptionsForNoPartitions(StaffMasqueradeTestCase):
+    """
+    Check that 'Learner' option is available if there are no groups or partitions
+    """
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_MASQUERADE': True})
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_ENROLLMENT_TRACK_USER_PARTITION': False})
+    def test_masquerade_options_enrollment_track(self):
+        self.verify_learner_masquerade_available(True)
+
+
+class TestMasqueradeOptionsForUserPartition(StaffMasqueradeTestCase):
+    """
+    Check that 'Learner' option is NOT available if there are no groups or partitions
+    """
+    @ patch.dict('django.conf.settings.FEATURES', {'ENABLE_ENROLLMENT_TRACK_USER_PARTITION': True})
+    @ patch.dict('django.conf.settings.FEATURES', {'ENABLE_MASQUERADE': True})
+    def test_masquerade_options_no_cohort(self):
+        self.verify_learner_masquerade_available(False)
```

Comment:
Have you used `ddt` much? If so and you just wanted to design your tests this way, alright. But if you haven't used it, we can chat about it - or you can go check current uses of it in platform. It would have been my first thought for how to organize these two tests. Only one test method, and could avoid the helper method.

Reply:
The patches on the tests create different setups, one with a partition, and one without. If I could have created one class that encompassed both tests, I would have used data injection.

Reply:
Fixed the tests to be one class using ddt to distinguish between them. Thanks!